doc/source/user_guide/codelist.rst

```diff
@@ -0,0 +1,244 @@
+pair defining the allowed **unit(s)** for the variable.
+
+This is an example for a valid entry in a variable codelist:
+
+.. code:: yaml
+
+   - Allowed variable name:
+     unit: A unit
+     description: A short description
+     <other attribute>: Some text, value, boolean or list (optional)
+
+The **unit** attribute is **required** and its value should be compatible with the
+`iam-units <https://github.com/iamconsortium/units>`_ package.
+
+The unit attribute can be:
+* a string -> one allowed unit for the variable
+* a list of strings -> a number of allowed units for the variable 
+* empty -> a *dimensionless* variable
+
+examples for all three options:
+
+    .. code:: yaml
+      
+      - Single unit variable:
+        unit: A single unit
+      - Multi unit variable:
+        unit: [unit 1, unit 2]
+      - Dimensionless variable:
+        unit:      
+
+While not strictly necessary a *description* attribute with a short description of the
+variable is encouraged. 
+
+In principle, *any* number of additional arbitrary named attributes are allowed. There
+are several attributes that affect the region-processing by the nomenclature package
+(see :ref:`region_aggregation_attributes`).
+
+.. _region_aggregation_attributes:
+
+Optional attributes for region aggregation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* By default, all variables are processed using the method
+  :meth:`pyam.IamDataFrame.aggregate_region`, which performs a simple summation of all
+  subregions.
+
+* Region aggregation for a particular variable can be skipped by using the attribute
+  *skip-region-aggregation* and setting it to ``true``:
+
+    .. code:: yaml
+
+       - Some Variable:
+         skip-region-aggregation: true
+
+  Setting *skip-region-aggregation* to ``true`` only skips the variable in question for
+  aggregation, if the variable is part of the provided data, it **is** used.
+
+* Any attributes which are arguments of
+  :meth:`aggregate_region() <pyam.IamDataFrame.aggregate_region>` will
+  be passed to that method. Examples include *method* and *weight*.
+
+* It is possible to rename the variable returned by the region processing using
+  a *region-aggregation* attribute, which must have a mapping of the target variable to
+  arguments of :meth:`aggregate_region() <pyam.IamDataFrame.aggregate_region>`.
+
+  This option can be used to compute several variables as part of the region-processing.
+  In the example below, the variable *Price|Carbon* is computed as a weighted average
+  using the CO2 emissions as weights, and in addition, the maximum carbon price within
+  each aggregate-region is added as a new variable *Price|Carbon (Max)*.
+
+    .. code:: yaml
+
+        - Price|Carbon:
+            unit: USD/t CO2
+            region-aggregation:
+              - Price|Carbon:
+                  weight: Emissions|CO2
+              - Price|Carbon (Max):
+                  method: max
+
+
+Guidelines and variable naming conventions
```

Comment:
I suggest to move this "Guideline" section above the section "Optional attributes for region aggregation", because it is "more important" (everyone should be aware of these guidelines, users don't necessarily need to know about the region-aggregation attributes...

Reply:
Switched the two.